src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,40 @@
      * Андрей - 600
      * София - 900</p>
      */
+
+    static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        if (LocalDate.parse(dateFrom, DATE_FORMAT)
+                .compareTo(LocalDate.parse(dateTo, DATE_FORMAT)) > 0) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        int sum = 0;
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Отчёт за период ")
+                .append(dateFrom).append(" - ")
+                .append(dateTo)
+                .append("\n");
+
+        for (String name: names) {
+            for (String employee: data) {
+                if (name.equals(employee.split(" ")[1])) {
```

Comment:
you call method split 5 times, it's better to create variable String[] and use it whenever you need to access element on particular index